src/components/beneficiarySelectionContent/beneficiarySelectionContent.tsx

```diff
@@ -0,0 +1,375 @@
+import React, { useState, useEffect } from 'react';
+import { View, FlatList, Text, TouchableOpacity } from 'react-native';
+import { useIntl } from 'react-intl';
+import { isArray, debounce } from 'lodash';
+
+import styles from './styles';
+import EStyleSheet from 'react-native-extended-stylesheet';
+
+import { useAppDispatch, useAppSelector } from '../../hooks';
+import { BeneficiaryModal, CheckBox, FormInput, IconButton, TextButton } from '../../components';
+import { Beneficiary } from '../../redux/reducers/editorReducer';
+import { lookupAccounts } from '../../providers/hive/dhive';
+import { TEMP_BENEFICIARIES_ID } from '../../redux/constants/constants';
+import {
+  removeBeneficiaries,
+  setBeneficiaries as setBeneficiariesAction,
+} from '../../redux/actions/editorActions';
+
+interface BeneficiarySelectionContentProps {
+  draftId: string;
+  setDisableDone: (value: boolean) => void;
+  powerDown?: boolean;
+  powerDownBeneficiaries?: Beneficiary[];
+  handleSaveBeneficiary?: (beneficiaries: Beneficiary[]) => void;
+  handleRemoveBeneficiary?: (beneficiary: Beneficiary) => void;
+}
+
+const BeneficiarySelectionContent = ({
+  draftId,
+  setDisableDone,
+  powerDown,
+  powerDownBeneficiaries,
+  handleSaveBeneficiary,
+  handleRemoveBeneficiary,
+}: BeneficiarySelectionContentProps) => {
+  const intl = useIntl();
+  const dispatch = useAppDispatch();
+
+  const beneficiariesMap = useAppSelector((state) => state.editor.beneficiariesMap);
```

Comment:
Map should be fed as prop instead of state from redux, because it is fetching same beneficiaries in power down screen which should not be the case.